cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -558,6 +563,12 @@ TEST_F(GcsIntegrationTest, GetFileInfoSelectorLimitedRecursion) {
                    }
                    return internal::Depth(info.path()) <= max_depth;
                  });
+    // Directories must appear without a trailing slash in the results.
+    std::transform(expected.begin(), expected.end(), expected.begin(),
+                   [](FileInfo const& info) {
+                     if (!info.IsDirectory()) return info;
+                     return Dir(internal::RemoveTrailingSlash(info.path()).to_string());
+                   });
```

Comment:
This is non-trivial enough to be factored out in a separate function, IMHO.

Reply:
Done.